packages/core/src/Queue.js

```diff
@@ -1,64 +1,106 @@
-/** @license MIT License (c) copyright 2010-2016 original author or authors */
-/** @author Brian Cavalier */
-/** @author John Hann */
+// Copied and modified from https://github.com/invertase/denque
+// MIT License
 
-// Based on https://github.com/petkaantonov/deque
+// These constants were extracted directly from denque's shift()
+// It's not clear exactly why the authors chose these particular
+// values, but given denque's stated goals, it seems likely that
+// they were chosen for speed/memory reasons.
+
+// Max value of _head at which Queue is willing to shink
+// its internal array
+const HEAD_MAX_SHRINK = 2
+
+// Min value of _tail at which Queue is willing to shink
+// its internal array
+const TAIL_MIN_SHRINK = 10000
```

Comment:
We should make performance tests to establish this value, and also ask the origin why this value was chosen.

Reply:
I feel like we can do these things without blocking this PR.  We're basically getting a 15% speed increase in zip for very little cost.

Reply:
Okay. Then we need to backlog that.

Reply:
Added it to the [1.x Backlog project](https://github.com/mostjs/core/projects/4).